src/controllers/notifications.js

```diff
@@ -71,6 +73,16 @@ const UnacceptableExpiryError = require('../errors/unacceptable-expiry-error')
 exports.post = function * postNotification () {
   const notification = yield requestUtil.validateBody(this, 'Notification')
   log.debug('Got notification: ' + JSON.stringify(notification))
+
+  const verifySignature = this.config.get('notifications.must_verify')
+  if (verifySignature) {
+    const result = model.verifySignature(notification, this.config)
+    if (!result.valid) {
+      log.warn('Signature verification failed: ' + result.error)
+      throw new UnprocessableEntityError('Notification failed signature verification')
```

Comment:
A little bit of context:
@justmoon earlier made a point that Ledger should retry on all errors (4xx and 5xx) because there might be a component, e.g., a firewall, between Ledger and Connector.  That component might respond to Ledger with an error for a reason Connector doesn't control.  If that happens, Ledger should keep retrying (with backoff), to be on the cautious side.  So that's why Ledger retries on both 4xx and 5xx today.  
But - I see @lumberj 's point, too.  In this case, Ledger failed to deliver a good notification with a good signature.  Ledger did something wrong, and it should receive an error.  
So I agree with @MatthewPhinney - this code is fine for now, but let's remember we'll have a message flooding problem if this error condition gets triggered (could you leave a comment in code?)  If this becomes a real problem, in the future, we can do what @MatthewPhinney suggested - throw a different error and treat it differently.